### PR TITLE
fix: debug print of egg versions doesn't handle corrupted eggs

### PR DESCRIPTION
**Pull request by grunwmar**
New if-else statement is added to check if exit code of previous Popen call is == 0 . Statement is added to insights.client.utilities.print_egg_version method.

CARD ID: CCT-406

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:

Originally, when unimportable egg was encountered, Insights client logged part of the traceback to the DEBUG log. I added a functionality which checks exit code of corresponding subprocess calling import and egg and if the code is not == 0, then it results to message ""path/to/egg: Corrupted egg." I was not able to capture exception from subprocess, so I choose this approach by using exit status. There is space for discussion whether there could happen another scenarios other than unimportable corrupted egg. Non-existing egg is already handled in code before.

### CODE REVIEW

**Review by m-horky — CHANGES_REQUESTED** [review]

File: insights/client/utilities.py
Comment (grunwmar) [reply insights/client/utilities.py]: @m-horky Thanks for feedback. Yes, that's why asked for discussion. I have changed the message to `%s: Could not read egg version` and logged `stdout` output. `stderr` is already redirected to `stdout` in `Popen` constructor.

File: insights/client/utilities.py
Comment (DuckBoss) [reply insights/client/utilities.py:311]: Also, just minor nitpicking, can you also update the commit title and pr title to remove the `wq` from `wqfix:...`?
`vim` strikes again :smiley:

**Review by DuckBoss — APPROVED** [review]
LGTM, thanks! :+1: 

:warning: Please wait for re-base before merging :warning:

**Comment by DuckBoss** [comment]
@xiangce this LGTM after recent rebase and my testing, can you merge this?

**Review by Glutexo — COMMENTED** [review]
We found an interesting 🐛 when the developer has the _insights-core_ package installed locally. This happens naturally, because the development [guide](https://insights-core-tutorials.readthedocs.io/en/latest/prep_tutorial_env.html#manual-setup-not-required-if-you-ran-setup-env-sh) tells us to run

```sh
$ pip install -e .[develop]
```

to install dependencies. The command, however, only installs the dependencies as a side-effect of installing the insights-core itself, putting in into PYTHONPATH. As a result, any subprocess spawned knows about the locally installed egg and falls back to it if a manually provided egg can’t be loaded.

```python
Popen([
    sys.executable,
    '-c',
    'from insights import package_info; print(\'%s-%s\' % (package_info[\'VERSION\'], package_info[\'RELEASE\']))'],
    env={'PYTHONPATH': egg, 'PATH': os.getenv('PATH')},
    stdout=PIPE,
    stderr=STDOUT
)
```

In this call, an egg is manually prepended to PYTHONPATH, forcing Python to (try to) load it first. If the egg is broken and can’t be imported, Python then imports the locally installed egg. The call is seemingly successful and

```python
logger.debug('%s: %s', egg, version)
```

prints then the version mislabeled.

**Comment by m-horky** [comment]
@Glutexo Please report that as a Jira ticket if it needs solving, we don't monitor closed PRs.

**Comment by Glutexo** [comment]
Yet, you noticed! 😎

**Comment by DuckBoss** [comment]
Thanks for noticing this and providing detailed information! It looks like our QE team has caught this case as well (refer to the jira). Please create a card for this and ping/assign me on it :)

**Pull request merged**
